src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -85,8 +85,13 @@ def __init__(self, feature_extractor: Union["SequenceFeatureExtractor", str], *a
             tokenizer ([`PreTrainedTokenizer`]):
                 The tokenizer that will be used by the pipeline to encode data for the model. This object inherits from
                 [`PreTrainedTokenizer`].
-            modelcard (`str` or [`ModelCard`], *optional*):
-                Model card attributed to the model for this pipeline.
+            chunk_length_ms (`int`, *optional*, defaults to 0):
+                The input length for in each chunk. If `0` then chunking is disabled (default). Only available for CTC
+                models.
+            stride_length_ms (`int`, *optional*, defaults to `chunk_length_ms / 6`):
```

Comment:
Also think it would be good to still have the option `"left"` and `"right"` for padding since later when we want to use pipelines also in "real" streaming mode (e.g. only buffers are sent to the pipeline) then ideally we would want to have a long left padding (since this is past speech), but a shorter right padding since there we have a trade-off between model performance and "real-time" display.

Reply:
Are you ok to leave that for subsequent PR, when the need arises ?

For streaming buffers, like microphone, chunking would be done elsewhere not here for instance since you don't want to wait to get large chunks to start inference, 50-100ms tops in terms of latency to get that "instantenous" feel. We may want to expose the `striding` mecanism still to get those "clean" reconstituion, but the chunking is IMO not here. For files or any static data there's not real overhead in terms of "real-time".

Personally, I would follow `torch` way here, meaning simple argument means both sides are the same, and tuple would mean (left, right). It means no extra argument (simpler) yet still doable when needed.

However I would leave that for a subsequent PR since I don't think it's that important right now (should be quite simple anyway).

Reply:
Definitely happy to leave that for a future PR, I'm just a bit worried here about backward compatibility.
I agree that for the use case enabled by the PR (**offline** chunked ASR), it's slightly better to have just a single "`stride_length_ms`" argument because it's not necessary to have different strides and fewer arguments is better for the user. 

However if we do need a different stride left and right then we would have to deprecate this argument in favor of `stride_length_left_s` and `stride_length_right_s` no? Which is what I'm a bit worried about.

We would need this however only once we implement the "online" streaming logic. Here is how I image it to work:

1.audio -> microphone -> buffer until `chunk_length_ms + stride_length_right_ms` is reached, *e.g.* `200ms + 50ms` -> forward array to pipelines *e.g.* `buffer[0:4000]` (with 16kHz) with `chunk_length_ms=0.2` and `stride_length_right_ms=0.05`, -> get output logits (or string) ->  cut 50ms of output logits (or string) -> store cut output logits or ids and display decoded text. -> emtpy buffer **except** right padding (`50ms`)
2. audio -> microphone buffer until `chunk_length_ms + stride_length_ms` is reached, *e.g.* `200ms + 50ms`, -> concatenate leftover buffer with new buffer -> forward array to pipelines (now including `stride_length_left_ms` (*e.g.* 1000ms)) so this time *e.g.* `buffer[0:7200]` -> cut 50ms of output logits (or string) -> overwrite stored output ids -> display 
3. ....
4. ....
5. This process would be repeated (the forwarded `buffer` would be starting  from `[0: ...]` until left stride + chunk size + right stride > received input so far. When this happens we would forward something like `[1000:20,000]` not starting from `[0: ...]`to the pipeline and then for the output we would have to retrieve the last stored output logits/ids and concatenate them with the currently model output logits/ids and display the decoded concatenated outputs.
6. ....
7. .... at some point the output logits or output ids are getting too large so that we'll need to completely throw away some stored logits, but I think this is only need at sizes of a minute or so which is probs too much to read anyways so it should be rather simple to just not display some words anymore.

Now the interesting question now is (and I'm open to both options here) whether the "cutting" logic of both the input (in case input > then left stride + chunk size + right stride) and the "cutting" logic of the output logits/ids (or strings if not possible or ugly to not decode in the pipeline) should or should not happen inside the pipeline. 

1.) If it's not the responsibility of the pipeline then IMO the pipeline should just receive an input (the cut buffer) and output ideally the output logits or the output string and something else is responsible for correctly cutting the logits. 
2.) If however we think it is the responsibility of the pipeline to cut both the input buffer and the output logits then I think we **do need** both left and right stride arguments. 

Having written down all of this, I actually think 1.) might make more sense here since there does seem to be a lot of logic which shouldn't necessarly be in `transformers` -> so I think I'm good with just having a single `stride_length_s` argument.

Curios to hear your thoughts here @Narsil @anton-l